ext/native.go

```diff
@@ -155,6 +156,14 @@ func (tp *nativeTypeProvider) FindStructType(typeName string) (*types.Type, bool
 	return tp.baseProvider.FindStructType(typeName)
 }
 
```

Comment:
The signature for `NativeTypes` takes an optional set of `any` values which are expected to be types; however, given this change enables tag handling in a very specific way and in a manner which could impact performance, could you consider introducing a functional option and filtering out the option values from the type values: 

```go
type NativeTypesOption func(*nativeTypeProvider) (*nativeTypeProvider, error)
```

You'd probably have to filter between options and types like so:

```go
nativeTypes := make([]any, 0, len(refTypes))
opts := []NativeTypesOption{}
for _, v := range refTypes {
  switch v := v.(type) {
    case NativeTypeOption:
      opts = append(opts, v)
    default:
      nativeTypes = append(nativeTypes, v)
  }
}
```

I would like to make sure that we're opting into the behavior change, and doing so in a way which won't impact performance.

Reply:
The struct tag parsing is now opt-in. Each `nativeType` struct now needs to store an additional flag to know how to retrieve the name for a field.